core/performance_test/peakflops/03_Cabana_peakflops.cpp

```diff
@@ -105,49 +103,55 @@ void movePx(data_t *__restrict__ a,  data_t *__restrict__ x0,
     }
 }
 
-//---------------------------------------------------------------------------//
-// Move function using the single particle index and slice syntax.
-template<typename SliceType>
-void moveSlices(SliceType a,  SliceType x0, SliceType x1, SliceType x2,
-               SliceType x3, SliceType x4, SliceType x5, SliceType x6,
-               SliceType x7, SliceType x8, SliceType x9, SliceType c,
-               long n, int num_struct )
+void //__attribute__ ((noinline))
+move_AoSoA(
+        ParticleList& a,  ParticleList& x0, ParticleList& x1, ParticleList& x2,
+        ParticleList& x3, ParticleList& x4, ParticleList& x5, ParticleList& x6,
+        ParticleList& x7, ParticleList& x8, ParticleList& x9, ParticleList& c,
+        long n, int num_struct )
 {
     long i;
     int s;
     int j;
 
     asm volatile ("# ax+c loop begin");
-    for(i = 0; i<n; i++)
-    {
         for ( s = 0; s < num_struct; ++s )
         {
-            for(j=s*VECLENTH; j<(s+1)*VECLENTH; j++)
+            for(i = 0; i<n; i++)
             {
-                x0(j) = a(j)*x0(j)+ c(j);
-                x1(j) = a(j)*x1(j)+ c(j);
-                x2(j) = a(j)*x2(j)+ c(j);
-                x3(j) = a(j)*x3(j)+ c(j);
-                x4(j) = a(j)*x4(j)+ c(j);
-                x5(j) = a(j)*x5(j)+ c(j);
-                x6(j) = a(j)*x6(j)+ c(j);
-                x7(j) = a(j)*x7(j)+ c(j);
-                x8(j) = a(j)*x8(j)+ c(j);
-                x9(j) = a(j)*x9(j)+ c(j);
+#pragma omp simd
+                for(j=0; j<VECLENTH; j++)
+                {
+                    auto _c = c.access(s).get<0>(j);
+                    auto _a = a.access(s).get<0>(j);
+
+                    x0.access(s).get<0>(j) = _a * x0.access(s).get<0>(j) + _c;
+                    x1.access(s).get<0>(j) = _a * x1.access(s).get<0>(j) + _c;
+                    x2.access(s).get<0>(j) = _a * x2.access(s).get<0>(j) + _c;
+                    x3.access(s).get<0>(j) = _a * x3.access(s).get<0>(j) + _c;
+                    x4.access(s).get<0>(j) = _a * x4.access(s).get<0>(j) + _c;
+                    x5.access(s).get<0>(j) = _a * x5.access(s).get<0>(j) + _c;
+                    x6.access(s).get<0>(j) = _a * x6.access(s).get<0>(j) + _c;
+                    x7.access(s).get<0>(j) = _a * x7.access(s).get<0>(j) + _c;
+                    x8.access(s).get<0>(j) = _a * x8.access(s).get<0>(j) + _c;
+                    x9.access(s).get<0>(j) = _a * x9.access(s).get<0>(j) + _c;
+                }
             }
-        }
     }
     asm volatile ("# ax+c loop end");
+    /* // TODO: do we need to re-enable this?
```

Comment:
What is this loop here for?

Reply:
@guangyechen we do want this, right?

Reply:
The motivation was to use the results of above loops so that compiler do not optimize them away.

Reply:
The `asm volatile` also does that, right?

Reply:
I don't know. If it does, then we can remove the 2nd loop.

Reply:
have you guys tried this out?

Reply:
Either way (removing or keeping it here) is fine here, because those `x0`-`x9` are used in the next call `moveSlicesWithAccess`, and the compiler will not optimize it away.

Reply:
@guangyechen @rfbird lets make a call here and resolve this one. If it stays please document in the comments why it is there.

Reply:
re-enabling so it's consistent with the other files